tests/bluetooth/bsim_bt/bsim_test_audio/src/has_client_test.c

```diff
@@ -39,9 +48,67 @@ static void preset_switch_cb(struct bt_has *has, uint8_t index)
 	SET_FLAG(g_preset_switched);
 }
 
+static void status_cb(struct bt_has *has, int err, void *user_data)
+{
+	if (user_data == bt_has_client_presets_read) {
+		printk("bt_has_client_presets_read ");
+	}
+
+	if (err) {
+		FAIL("%s: err %d\n", __func__, err);
+	} else {
+		printk("%s: err %d\n", __func__, err);
+	}
+}
+
+static bool match_preset_record(const struct bt_has_preset_record_val *value,
+				enum bt_has_properties expected_properties,
+				const char *expected_name)
+{
+	bool result;
+
+	result = value->properties == expected_properties &&
+		 !strcmp(value->name, expected_name);
+
+	if (result == false) {
+		FAIL("mismatch 0x%02x %s vs 0x%02x %s expected\n", value->properties, value->name,
+		     expected_properties, expected_name);
+	}
+
+	return result;
+}
+
+static void preset_event_cb(struct bt_has *has, enum bt_has_preset_event_type type, uint8_t index,
+			    const void *data, bool is_last)
+{
+	if (type == BT_HAS_PRESET_RECORD_READ_RSP) {
+		if (index == test_preset_index_1) {
+			if (!match_preset_record(data, test_preset_properties,
+			    test_preset_name_1)) {
+				SET_FLAG(g_preset_validation_failed);
+			}
+
+			SET_FLAG(g_preset_1_found);
+		} else if (index == test_preset_index_5) {
+			if (!match_preset_record(data, test_preset_properties,
+			    test_preset_name_5)) {
+				SET_FLAG(g_preset_validation_failed);
+			}
+
+			SET_FLAG(g_preset_5_found);
+		} else {
+			FAIL("unexpected index 0x%02x", index);
+
+			SET_FLAG(g_preset_validation_failed);
```

Comment:
What's the reasoning for setting a failing flag and then `FAIL` on it later (or even before in the last case here), instead of just calling `FAIL`?

Reply:
Yes, it makes no sense at all.